pkg/steps/multi_stage.go

```diff
@@ -145,6 +154,13 @@ func (s *multiStageTestStep) generatePods(steps []api.TestStep) ([]coreapi.Pod,
 		if owner := s.jobSpec.Owner(); owner != nil {
 			pod.OwnerReferences = append(pod.OwnerReferences, *owner)
 		}
+		if s.artifactDir != "" && step.ArtifactDir != "" {
```

Comment:
Why do we need to check both?

Reply:
`s.artifactDir`: the user has requested artifact gathering via `--artifact-dir`, which is where this value ultimately comes from. There is nowhere to output to if that's not the case.

`step.ArtifactDir`: the step has declared an artifact directory. There is no reason to add the mount if it doesn't.

Reply:
The pod step (unbeknownst to me when I wrote the multi-stage code) uses the exact same logic: https://github.com/openshift/ci-tools/blob/36ac9f630ea7308baacfba2779564d6f5cf544eb/pkg/steps/pod.go#L131.

Reply:
Although maybe we should add the volume even if `--artifact-dir` is not specified? If a test has `artifact_dir`, I would think it requires that directory to exist.

But… it has worked so far, so /shrug?